dev/test/bulk-writer.ts

```diff
@@ -0,0 +1,502 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {expect} from 'chai';
+import {Status} from 'google-gax';
+
+import {Firestore, setLogFunction, Timestamp, WriteResult} from '../src';
+import {BulkWriter} from '../src/bulk-writer';
+import {Deferred} from '../src/util';
+import {
+  ApiOverride,
+  create,
+  createInstance,
+  document,
+  remove,
+  response,
+  set,
+  update,
+  updateMask,
+  verifyInstance,
+} from './util/helpers';
+
+// Change the argument to 'console.log' to enable debug output.
+setLogFunction(() => {});
+
+const PROJECT_ID = 'test-project';
+
+interface BulkWriterRequest {
+  writes: Array<{[key: string]: {} | undefined}>;
+}
+
+interface BulkWriterResponse {
+  writeResults: Array<{[key: string]: {} | null}>;
+  status: Array<{[key: string]: {}}>;
+}
```

Comment:
The way these two types are defined removes all type safety from the tests. Please try to use the Protobuf types as defined in https://github.com/googleapis/nodejs-firestore/blob/master/dev/protos/firestore_v1_proto_api.d.ts

Reply:
done.